plugins/woocommerce-blocks/packages/components/text-input/validated-text-input.tsx

```diff
@@ -117,29 +121,39 @@ const ValidatedTextInput = forwardRef<
 
 				if (
 					inputObject.checkValidity() &&
-					customValidationRef.current( inputObject )
+					customValidationRef.current( inputObject ) &&
+					errorsHidden
 				) {
 					clearValidationError( errorIdString );
 					return;
 				}
 
-				setValidationErrors( {
-					[ errorIdString ]: {
-						message: getValidityMessageForInput(
-							label,
-							inputObject,
-							customValidityMessage
-						),
-						hidden: errorsHidden,
-					},
-				} );
+				if ( ! errorsHidden ) {
+					showValidationError( errorIdString );
+				}
```

Comment:
Showing errors on blur instead of relaying on creating a new error

Reply:
Previously this hook would clear up all errors and generate news ones with hidden or not based on `errorsHidden`, this didn't reveal errors coming from other places, so the fix above fixes that.